user-guide/Troubleshooting/Procedures/Cassandra_expired_blocking_SSTables.md

```diff
@@ -0,0 +1,63 @@
+---
+uid: Cassandra_expired_blocking_SSTables
+---
+
+# Expired SSTables blocking other tables from being dropped
+
+In case the only data SSTables in Cassandra contain is tombstones, and the tables are therefore expired, they could block other tables that are also expired from being dropped. This could result in various issues, such as an increase in the disk usage of the Cassandra data partition, or memory leak behavior in the TWCS keyspaces (*trend_data_medium*, *trend_data_short*, and *trend_data_long*).
+
+You can check if you are indeed encountering this issue using the Cassandra tools, for example with the following commands (on Linux):
+
+```txt
+yum install cassandra-tools
+
+sudo systemctl stop cassandra
+
+sstableexpiredblockers zm1_cc_trend_data_short trend_data_short
+
+for f in *Data.db; do meta=$(sudo sstablemetadata $f); echo -e "Max:" $(date --date=@$(echo "$meta" | grep Maximum\ time | cut -d" " -f3| cut -c 1-10) '+%m/%d/%Y') "Min:" $(date --date=@$(echo "$meta" | grep Minimum\ time | cut -d" " -f3| cut -c 1-10) '+%m/%d/%Y') $(echo "$meta" | grep droppable) ' \t ' $(ls -lh $f | awk '{print $5" "$6" "$7" "$8" "$9}'); done | sort
+```
+
+With the output of the sstableexpiredblockers, you can see which tables are blocking others from being dropped. If the percentage of tombstones in the SSTable is higher than 1, you can assume that all data is tombstone data.
+
+To resolve this issue, we recommend enabling the *unsafe_aggressive_sstable_expiration* option for the Cassandra database. This is something that should only be done to troubleshoot this issue, because it is considered a risky option otherwise. The data in the expired SSTables will be dropped without being checked, but as it consists entirely of tombstone data in this case, this is not a problem.
```

Comment:
I don't understand why you would only enable this option for troubleshooting or why it is a risky option otherwise. @joncorral can you elaborate?

Reply:
I added that remark based on the Apache Cassandra documentation that says the following:

This is a potentially risky option that can lead to data loss or deleted data re-appearing, going beyond what unchecked_tombstone_compaction does for single sstable compaction. 

That is why I would not use this option if not facing this issue.

Reply:
Then I would say:
'Enabling this option on a table could lead to inconsistencies (data loss or re-appearing) when deletes/updates are done on the data in TWCS tables. In general, deletes and updates should be avoided on tables with TWCS and you should always rely on TTL as this could result in the blocking tables issue.'

Reply:
That looks very clear and understandable. Thanks Michiel